tests/check_framework/test_translation.py

```diff
@@ -0,0 +1,47 @@
+from django.conf import settings
+from django.core.checks.translation import check_setting_language_code
+from django.test import SimpleTestCase
+from django.test.utils import override_settings
+
+
+class TranslationCheckTests(SimpleTestCase):
+
+    @override_settings(ROOT_URLCONF='check_framework.translation.valid_language_code_format_ll_only')
+    def test_valid_language_code_format_ll_only(self):
+        settings.LANGUAGE_CODE = "en"
+        result = check_setting_language_code(None)
+        self.assertEqual(len(result), 0)
+
+    @override_settings(ROOT_URLCONF='check_framework.translation.invalid_language_code_format_ll_only')
+    def test_invalid_language_code_format_ll_only(self):
+        invalid_codes = ["EN", "eN"]
+        for code in invalid_codes:
```

Comment:
I'm not sure there's much value in testing two invalid codes instead of just one.

Reply:
Fair point, test one value in `TranslationCheckTests`. Added `TranslationLanguageCodeRegexTest` to test regex using `subTest()`